datafusion/core/src/execution/context.rs

```diff
@@ -1405,6 +1405,15 @@ pub fn default_session_builder(config: SessionConfig) -> SessionState {
     SessionState::with_config_rt(config, Arc::new(RuntimeEnv::default()))
 }
 
+impl Default for SessionState {
```

Comment:
I am worried that providing a Default for SessionState might allow subtle bugs, specifically accidentally running queries or other operations without properly threading through the configuration or runtime. I believe that is why there is no `SessionState::new()` or `Default` 

🤔  this context would be nice to add as a comment to SessonState

Reply:
Yes, it there are many options to set in `SessionState `,  i just want to write less code when construct SessionState see there are already default in `SessionConfig` and `RuntimeEnv` 😂 ， it make sense  let user point what conf they want.

Reply:
I think we should remove the `Default` impl and instead have a function for tests (`test_session_state()` or similar) to avoid misuse of this API. 

Here is a PR to add some comments about the rationale for not having a `Default` impl:  https://github.com/apache/arrow-datafusion/pull/7654